400-claims.md

```diff
@@ -0,0 +1,197 @@
+
+### Parameters
+
+The parameter data stored in a claim data is _the resolved key/value pairs_ that result from the following transformation:
+
+- The values supplied by the user are validated by the rules specified in the `bundle.json` file
+- The output of this operation is a set of key/value pairs in which:
+  - Valid user-supplied values are presented
+  - Default values are supplied for all parameters where `defaultValue` is provided and no user-supplied value overrides this
+
+### How is the Claim Used
+
+The claim is used to inform any CNAB tooling about how to address a particular installation. For example, given the claim record, a package manager that implements CNAB should be able to:
+
+- List the _names_ of the installations, given a _bundle name_
+- Given an installation's _name_, return the _bundle info_ that is installed under that name
+- Given an installation _name_ and a _bundle_, generate a _bundle info_.
+  - This is accompanied by running the `install` path in the bundle
+- Given an installation's _name_, replace the _bundle info_ with updated _bundle info_, and update the revision with a new ULID, and the modified timestamp with the current time. This is an upgrade operation.
+  - This is accompanied by running the `upgrade` path in the bundle
+- Given an installation's name, mark the claim as deleted.
+  - This is accompanied by running the `uninstall` path in the bundle
+  - XXX: Do we want to allow the implementing system to remove the claim from its database (e.g. helm delete --purge) or remain silent on this matter?
+
+To satisfy these requirements, implementations of a CNAB package manager are expected to be able to store and retrieve state information. However, note that nothing in the CNAB specification tells _how or where_ this state information is to be stored. It is _not a requirement_ to store that state information inside of the invocation image. (In fact, this is discouraged.)
+
+## Injecting Claim Data into an Invocation Image
+
+Complaint CNAB implementations MUST conform to this section.
+
+The claim is produced outside of the CNAB package. The following claim data is injected
+into the invocation container at runtime:
+
+- `$CNAB_INSTALLATION_NAME`: The value of `claim.name`.
+- `$CNAB_BUNDLE_NAME`: The name of the present bundle.
+- `$CNAB_ACTION`: The action to be performed (install, upgrade, ...)
+- `$CNAB_REVISION`: The ULID for the present release revision. (On upgrade, this is the _new_ revision)
+
+> Credential data, which is also injected into the invocation image, is _not_ managed by the claim system. Rules for injecting credentials are found in [the bundle runtime definition](103-bundle-runtime.md).
+
+Parameters that are declared with an `env` key in the `destination` object, their values will be injected as environment variables according to the name specified. Likewise, files will be injected if `path` is set on `destination`.
```

Comment:
First sentence may benefit from rewording.  Perhaps something like: `Parameters declared with an env key in the destination object will be injected as environment variables according to the name specified.`

Reply:
Oh, man... that sentence is just embarrassing. Sorry about that.